### PR TITLE
t: Use explicit wait functions where needed

### PR DESCRIPTION
**Pull request by okurz**
To not rely on a global Selenium timeout in test functions we should use
our explicit wait functions where they are needed.

### CODE REVIEW

File: t/ui/16-tests_job_next_previous.t [hunk 1]
```diff
@@ -104,13 +104,12 @@ like(
 # trigger job next and previous for current job
 $driver->title_is('openQA', 'on main page');
 $driver->find_element_by_link_text('All Tests')->click();
-$driver->find_element('[href="/tests/99946"]')->click();
+wait_for_element(selector => '[href="/tests/99946"]')->click();
 $driver->find_element_by_link_text('Next & previous results')->click();
-wait_for_ajax();
```
Comment (okurz) [inline t/ui/16-tests_job_next_previous.t]: wait_for_ajax is actually smart. We could combine this with wait_for_element or wait_until. wait_for_ajax should return if it actually did something

Comment (Martchus) [reply t/ui/16-tests_job_next_previous.t]:
```
diff --git a/t/lib/OpenQA/SeleniumTest.pm b/t/lib/OpenQA/SeleniumTest.pm
index bbd14a7da..63c922d68 100644
--- a/t/lib/OpenQA/SeleniumTest.pm
+++ b/t/lib/OpenQA/SeleniumTest.pm
@@ -185,6 +185,7 @@ sub _default_check_interval {
 sub wait_for_ajax {
     my $check_interval = _default_check_interval(shift);
     my $timeout        = 60 * 5;
+    my $slept = 0;
 
     while (!$_driver->execute_script('return window.jQuery && jQuery.active === 0')) {
         if ($timeout <= 0) {
@@ -194,8 +195,10 @@ sub wait_for_ajax {
 
         $timeout -= $check_interval;
         sleep $check_interval;
+        $slept = 1;
     }
     pass("Wait for jQuery successful");
+    return $slept;
 }
 
 sub disable_bootstrap_animations {
@@ -328,7 +331,7 @@ sub wait_until {
             return 0;
         }
         $timeout -= $check_interval;
-        sleep $check_interval;
+        wait_for_ajax or sleep $check_interval;
     }
 }
```

Comment (okurz) [reply t/ui/16-tests_job_next_previous.t]: Done in https://github.com/os-autoinst/openQA/pull/2765/commits/5e75a80bfc171a433588230cf185961a10a9c91f

**Comment by okurz** [comment]
I consider this also pair-reviewed with mkittler and mergeable.

**Pull request merged**